build_tools/cmake/iree_copts.cmake

```
@@ -131,6 +131,54 @@ iree_select_compiler_opts(IREE_DEFAULT_COPTS
 set(IREE_DEFAULT_LINKOPTS "${ABSL_DEFAULT_LINKOPTS}")
 set(IREE_TEST_COPTS "${ABSL_TEST_COPTS}")
 
+#-------------------------------------------------------------------------------
+# Sanitizer configurations
+#-------------------------------------------------------------------------------
+
+include(CheckCXXCompilerFlag)
+if(${IREE_ENABLE_ASAN})
+  if(NOT ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+    message(FATAL_ERROR "IREE_ENABLE_ASAN requires Debug build")
+  endif()
+  set(CMAKE_REQUIRED_FLAGS "-fsanitize=address")
+  check_cxx_compiler_flag(-fsanitize=address COMPILER_SUPPORTS_ASAN)
+  unset(CMAKE_REQUIRED_FLAGS)
+  if(${COMPILER_SUPPORTS_ASAN})
+    string(APPEND CMAKE_C_FLAGS_DEBUG " -fsanitize=address")
+    string(APPEND CMAKE_CXX_FLAGS_DEBUG " -fsanitize=address")
+  else()
+    message(FATAL_ERROR "The compiler does not support address sanitizer")
```

The checks performed above are not exhaustive, so this isn't necessarily true. MSVC has some ASAN support and would use different flags. Maybe there's a configuration from another project that we could reference? https://devblogs.microsoft.com/cppblog/addresssanitizer-asan-for-windows-with-msvc/#turning-on-asan-for-windows-cmake-projects . Clang/GCC only is fine to start with, but can you tweak the error message slightly?

```suggestion
    message(FATAL_ERROR "The compiler does not support or is missing configuration for address sanitizer")
```